chia/wallet/wallet_state_manager.py

```diff
@@ -1417,8 +1417,12 @@ async def _add_coin_states(
                             additions = [state.coin for state in children]
                             if len(children) > 0:
                                 fee = 0
-
+                                known_puzzle_hash: bool = await self.puzzle_store.puzzle_hash_exists(
```

Comment:
Yes, that is the assumption. This is a double-check since we want to do this expensive check only if it is necessary.

Reply:
Right now it doesn't help to avoid the expensive check though because you only check against `known_puzzle_hash` after you already fetched and processed the coin spend, right?

If you really want to make this potentially avoid the expensive check it should rather be like below i guess?

```python
if known_puzzle_hash and coin_spend is None:
``` 

But again, can you think about any situation where we would end up here without already knowing this coin_state's puzzle hash belongs to us? Just want to make sure we don't add another DB lookup for every coin only to "double check" if its at the end not really required.